omod-1.10/src/test/java/org/openmrs/module/attachments/rest/AttachmentRestController1_10Test.java

```diff
@@ -409,4 +473,80 @@ public void getAttachmentBytes_shouldDownloadFile() throws Exception {
 		Assert.assertEquals(downloadResponse.getHeader("File-Ext"), fileExtension);
 		
 	}
+	
+	@Test
+	public void doSearch_shouldGetAttachmentsWithMimeTypesAndContentFamilySet() throws Exception {
+		
+		// Setup
+		String fileCaption = "Test file caption";
+		String mimeType = "application/octet-stream";
+		String fileExtension = "dat";
+		String fileName = "testFile." + fileExtension;
+		
+		Patient patient = createBasicPatient();
+		MockMultipartHttpServletRequest uploadRequest = newUploadRequest(getURI());
+		MockMultipartFile file = new MockMultipartFile("file", fileName, mimeType, randomData);
+		
+		uploadRequest.addFile(file);
+		uploadRequest.addParameter("patient", patient.getUuid());
+		uploadRequest.addParameter("fileCaption", fileCaption);
+		
+		// Save attachment
+		deserialize(handle(uploadRequest));
+		
+		// Search attachments for patient
+		MockHttpServletRequest request = request(RequestMethod.GET, getURI());
+		request.addParameter("patient", patient.getUuid());
+		SimpleObject response = deserialize(handle(request));
+		LinkedHashMap<String, String> result = (LinkedHashMap<String, String>) ((ArrayList<LinkedHashMap>) response
+		        .get("results")).get(0);
+		
+		// Verify
+		Assert.assertEquals("application/octet-stream", result.get("bytesMimeType"));
+		Assert.assertEquals(ContentFamily.OTHER.toString(), result.get("bytesContentFamily"));
+	}
+	
+	private Patient createBasicPatient() {
+		Patient patient = new Patient();
+		
+		PersonName pName = new PersonName();
+		pName.setGivenName("Tom");
+		pName.setMiddleName("E.");
+		pName.setFamilyName("Patient");
+		patient.addName(pName);
+		
+		PersonAddress pAddress = new PersonAddress();
+		pAddress.setAddress1("123 My street");
+		pAddress.setAddress2("Apt 402");
+		pAddress.setCityVillage("Anywhere city");
+		pAddress.setCountry("Some Country");
+		Set<PersonAddress> pAddressList = patient.getAddresses();
+		pAddressList.add(pAddress);
+		patient.setAddresses(pAddressList);
+		patient.addAddress(pAddress);
+		// patient.removeAddress(pAddress);
+		
+		patient.setBirthdateEstimated(true);
+		patient.setBirthdate(new Date());
+		patient.setBirthdateEstimated(true);
+		patient.setDeathDate(new Date());
+		patient.setCauseOfDeath(new Concept());
+		patient.setGender("male");
+		
+		List<PatientIdentifierType> patientIdTypes = Context.getPatientService().getAllPatientIdentifierTypes();
+		
+		PatientIdentifier patientIdentifier = new PatientIdentifier();
+		patientIdentifier.setIdentifier("123-0");
+		patientIdentifier.setIdentifierType(patientIdTypes.get(0));
+		patientIdentifier.setLocation(new Location(1));
+		patientIdentifier.setPreferred(true);
+		
+		Set<PatientIdentifier> patientIdentifiers = new LinkedHashSet<>();
+		patientIdentifiers.add(patientIdentifier);
+		
+		patient.setIdentifiers(patientIdentifiers);
+		
+		return Context.getPatientService().savePatient(patient);
+	}
```

Comment:
Why don't you use a patient from the standard test dataset?

Reply:
Sorry. @gitcliff must have forced pushed again and deleted my commits. This shouldn't be here.